apps/sequencer/src/providers/eth_send_utils.rs

```diff
@@ -205,6 +211,133 @@ pub async fn get_serialized_updates_for_network(
     Ok(serialized_updates)
 }
 
+pub struct BatchOfUpdatesToProcess {
+    pub net: String,
+    pub provider: Arc<Mutex<RpcProvider>>,
+    pub provider_settings: blocksense_config::Provider,
+    pub updates: BatchedAggegratesToSend,
+    pub feed_type: Repeatability,
+    pub feeds_config: Arc<RwLock<HashMap<u32, FeedConfig>>>,
+    pub transaction_retry_timeout_secs: u64,
+    pub transaction_retries_count_limit: u64,
+    pub retry_fee_increment_fraction: f64,
+}
+
+pub async fn create_and_collect_relayers_futures(
+    collected_futures: &FuturesUnordered<JoinHandle<Result<(), Error>>>,
+    feeds_metrics: Arc<RwLock<FeedsMetrics>>,
+    provider_status: Arc<RwLock<HashMap<String, ProviderStatus>>>,
+    relayers_recv_channels: HashMap<String, UnboundedReceiver<BatchOfUpdatesToProcess>>,
+) {
+    for (net, chan) in relayers_recv_channels.into_iter() {
+        let feed_metrics_clone = feeds_metrics.clone();
+        let provider_status_clone = provider_status.clone();
+        let relayer_name = format!("relayer_for_network {net}");
+        collected_futures.push(
+            tokio::task::Builder::new()
+                .name(relayer_name.clone().as_str())
+                .spawn(async move {
+                    loop_processing_batch_of_updates(
+                        net,
+                        relayer_name,
+                        feed_metrics_clone,
+                        provider_status_clone,
+                        chan,
+                    )
+                    .await;
+                    Ok(())
+                })
+                .expect("Failed to spawn metrics collector loop!"),
+        );
+    }
+}
+
+pub async fn loop_processing_batch_of_updates(
+    net: String,
+    relayer_name: String,
+    feeds_metrics: Arc<RwLock<FeedsMetrics>>,
+    provider_status: Arc<RwLock<HashMap<String, ProviderStatus>>>,
+    mut chan: UnboundedReceiver<BatchOfUpdatesToProcess>,
+) {
+    tracing::info!("Starting {relayer_name} loop...");
+
+    loop {
```

Comment:
This loop is never ending. You can add a TODO to create a `termination reason` pattern in the future. This way we will have more control for the workers.